evalml/tests/automl_tests/test_automl.py

```diff
@@ -5454,3 +5454,60 @@ def test_get_ensembler_input_pipelines(X_y_binary, AutoMLTestEnv):
     error_text = "Pipeline ID 500 is not a valid ensemble pipeline"
     with pytest.raises(ValueError, match=error_text):
         automl.get_ensembler_input_pipelines(500)
+
+
+@pytest.mark.parametrize(
+    "problem_type",
+    [
+        ProblemTypes.TIME_SERIES_REGRESSION,
+        ProblemTypes.TIME_SERIES_MULTICLASS,
+        ProblemTypes.TIME_SERIES_BINARY,
+    ],
+)
+def test_automl_passes_known_in_advance_pipeline_parameters_to_all_pipelines(
```

Comment:
One tech debt issue with the algorithm upgrade is separating pipeline building logic out of search and removing pipeline building out of tests as well (#2868 tracks this). Do you think this test better belongs in `test_iterative_algorithm.py`? You can just test against `allowed_pipelines` to see if the pipelines generated match what you would expect.  Once I get to #2691 I can add a test in `DefaultAlgorithm` as well!

Reply:
Thanks for bringing this up! The known-in-advance implementation requires us to set the parameters for SelectColumns correctly for all pipelines. That will have to hold for all automl algorithms - iterative, default, whatever we come up with in the future. Because of that, I thought it'd be cleaner to have a test in `test_automl` and then parametrize over automl algorithms (when default algorithm supports time series) as opposed to a test in every automl test file.

What do you think?

Reply:
Gotcha, that make a whole lot of sense to me. Thanks for explaining, the current test is great!